src/intervals/reduction.jl

```diff
@@ -0,0 +1,45 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+"""
+The file contains the reduction functions stated in the section 12.12.12 with correctly rounded results.
+r is the rounding direction which takes 0.0 for RoundingToZero , 0.5 for RoundingNearest , +Inf for RoundingUp and -Inf for RoundingDown.
+"""
+function mpfr_vector_sum(v :: Vector{T} , r :: Float64) where T
+    n = length(v)
+    sum = BigFloat(0)
```

Comment:
You are not taking care with the precision of the `BigFloat`. In order for it to guarantee a correctly-rounded result, section 12.13.5 specifies the required precision.

Reply:
Sorry but I couldn't see the exact figure as the precision in the document. Rather it suggests a new format datatype `accumulator`.